src/main/java/gov/nasa/cumulus/CNMResponse.java

```diff
@@ -98,14 +101,46 @@ public static JsonObject getResponseObject(String exception) {
 		return response;
 	}
 
-	public static String generateOutput(String inputCnm, String exception, JsonObject granule){
+	public static String generateOutput(String inputCnm, String exception, JsonObject granule, JsonObject  inputConfig){
 		//convert CNM to GranuleObject
 		JsonElement jelement = new JsonParser().parse(inputCnm);
 		JsonObject inputKey = jelement.getAsJsonObject();
-
-		JsonElement sizeElement = inputKey.get("product").getAsJsonObject().get("files").getAsJsonArray().get(0).getAsJsonObject().get("size");
-
-		inputKey.remove("product");
+		String distribute_url = inputConfig.getAsJsonPrimitive("distribution_endpoint").getAsString();
+
+		JsonArray granuleFiles = granule.getAsJsonArray("files");
+		// build product.files off input granule's files
+		JsonArray productFiles =  new JsonArray();
+		granuleFiles.forEach((JsonElement e) -> {
+			JsonObject f = new JsonObject();
+			//type
+			f.addProperty("type", e.getAsJsonObject().getAsJsonPrimitive("type").getAsString());
+			// subtype : skip
+			// name
+			f.addProperty("name", e.getAsJsonObject().getAsJsonPrimitive("name").getAsString());
+			// uri
+			String filename = e.getAsJsonObject().getAsJsonPrimitive("filename").getAsString();
+			Matcher m = getSourceBucketAndKey(filename);
+			if (m.find()) {
+				String sourceBucket = m.group(1);
+				String key = m.group(2);
+				f.addProperty("uri",distribute_url + key);
```

Comment:
This strips off the bucket but we need the bucket in the distribution URL.

https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/MODIS_A-JPL-L2P-v2019.0/20020704000015-JPL-L2P_GHRSST-SSTskin-MODIS_A-N-v02.0-fv01.0.nc

Simply replacing s3:// with https://archive.podaac.earthdata.nasa.gov/ should be sufficient

Reply:
Also, please move all the code that adds product to inside the 
SUCCESS check

`if (granule != null && response.get("status").getAsString().equals("SUCCESS")) {`

Because if ingest failed for any reason it doesn't make sense to return links to data.

Reply:
All above is done.
Also for failure case, the product object will not be in the response. Since if it if FAILURE, the product object won't be rebuilt based on input granule object.